### PR TITLE
Opgops 2211

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

File: roles/awscommon/tasks/vpc-facts.yml [hunk 1]
```diff
@@ -8,3 +8,43 @@
...
+  ec2_vpc_net_facts:
+    filters:
+      "tag:Stack": "{{ vpc_name }}"
+  register: vpc_data
+  tags:
+    - always
+
+- name: Set VPC id
+  set_fact:
+    vpc_id: "{{ vpc_data.vpcs.0.id }}"
```
Comment (Azulinho) [inline roles/awscommon/tasks/vpc-facts.yml]: can you add an example of what you get back on that vpc_data.vpcs array ?

Comment (Azulinho) [reply roles/awscommon/tasks/vpc-facts.yml]: do you always want the first element?

Comment (brettminnie) [reply roles/awscommon/tasks/vpc-facts.yml]: Sure, wrt example and yes, we get back an array with one element

File: roles/awscommon/tasks/vpc-facts.yml [hunk 2]
```diff
@@ -8,3 +8,62 @@
...
+#                "state": "available",
+#                "tags": {
+#                    "Environment": "foo",
+#                    "Name": "dev-vpc.dev.foo.opg.digital",
+#                    "Project": "foo",
+#                    "Stack": "dev-vpc"
+#                }
+#            }
+#        ]
+
```
Comment (Azulinho) [inline roles/awscommon/tasks/vpc-facts.yml:38]: this is really helpfull ^

File: roles/awscommon/tasks/vpc-facts.yml [hunk 3]
```diff
@@ -8,3 +8,62 @@
...
+- name: Get list of subnets
+  ec2_vpc_subnet_facts:
+    filters:
+      vpc-id: "{{ vpc_id }}"
+  register: subnet_data
+  check_mode: no
+  tags:
+    - always
+
```
Comment (Azulinho) [inline roles/awscommon/tasks/vpc-facts.yml:53]: fancy doing the same example ^ you did for the vpcs ?

File: roles/awscommon/tasks/vpc-facts.yml [hunk 4]
```diff
@@ -8,3 +8,62 @@
   register: vpc_instance_facts
   tags:
     - always
+
+- name: get VPC data
```
Comment (carinadigital) [inline roles/awscommon/tasks/vpc-facts.yml]: We have other tasks called this. How about `get VPC net data`

**Review by carinadigital — APPROVED** [review]

File: roles/awscommon/tasks/vpc-facts.yml (hunk 4 above)
Comment (brettminnie) [reply roles/awscommon/tasks/vpc-facts.yml]: I'll rename

**Comment by brettminnie** [comment]
test

**Comment by brettminnie** [comment]
test

**Pull request merged**